src/libYARP_OS/src/BottleImpl.cpp

```diff
@@ -805,7 +805,7 @@ bool StoreVocab::writeRaw(ConnectionWriter& writer)
 ConstString StoreDouble::toString() const
 {
     char buf[512];
-    sprintf(buf, "%g", x);
+    sprintf(buf, "%.*g", DBL_DIG, x);
```

Comment:
Per [`iwyu`](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md), I would add an `#include <cfloat>` to this file to avoid unexpected breakages in the future.

Reply:
Done!